### PR TITLE
add eip158 to Byzantium transition test

### PR DESCRIPTION
**Pull request by ayrat555**

### CODE REVIEW

File: apps/blockchain/lib/blockchain/chain.ex [hunk 1]
```diff
@@ -144,6 +144,7 @@ defmodule Blockchain.Chain do
       iex> Blockchain.Chain.test_config("Frontier").name
       "Frontier (Test)"
   """
+  # credo:disable-for-next-line
```
Comment (InoMurko) [inline apps/blockchain/lib/blockchain/chain.ex]: how is this a fix?

Comment (ayrat555) [reply apps/blockchain/lib/blockchain/chain.ex]: the function consists of only one `case` statement with many variants. I think that's ok.

Comment (ayrat555) [reply apps/blockchain/lib/blockchain/chain.ex]: what do you suggest?

Comment (InoMurko) [reply apps/blockchain/lib/blockchain/chain.ex]: it doesn't like the huge case, so if you split this into pattern matching it'll be ok.
```
def test_config(hardfork) when is_binary(hardfork) do
  config = EVM.Configuration.hardfork_config(hardfork)
  do_test_config(hardfork, config)
end
defp do_test_config("Frontier", config), do: load_chain(:frontier_test, config)
defp do_test_config("Homestead", config), do: load_chain(:homestead_test, config)
defp do_test_config("TangerineWhistle", config), do: load_chain(:eip150_test, config)
defp do_test_config("SpuriousDragon", config), do: load_chain(:eip161_test, config)
defp do_test_config("Byzantium", config), do: load_chain(:byzantium_test, config)
defp do_test_config("Constantinople", config), do: load_chain(:constantinople_test, config)
defp do_test_config("ByzantiumToConstantinopleAt5", config), do: load_chain(:byzantium_to_constantinople_transition_test, config)
defp do_test_config("EIP158ToByzantiumAt5", config), do: load_chain(:eip158_to_byzantium_transition_test, config)
defp do_test_config("HomesteadToDaoAt5", config), do: load_chain(:dao_hardfork_test, config)
```

Comment (ayrat555) [reply apps/blockchain/lib/blockchain/chain.ex]: IMHO it looks much worse

Comment (InoMurko) [reply apps/blockchain/lib/blockchain/chain.ex]: well there are other ways. you can cast the binary to atom and call the function that would give you back the function, for example.
having huge functions with cases that are likely to grow might not be the best idea.
good read anyhow: http://erlang.org/pipermail/erlang-questions/2008-March/033403.html

Comment (InoMurko) [reply apps/blockchain/lib/blockchain/chain.ex]:
```
def test_config(hardfork) when is_binary(hardfork) do
    config = EVM.Configuration.hardfork_config(hardfork)
    test = get_test(hardfork)
    load_chain(test, config)
  end

  defp get_test("Frontier"), do: :frontier_test
  defp get_test("Homestead"), do: :homestead_test
  defp get_test("TangerineWhistle"), do: :eip150_test
  defp get_test("SpuriousDragon"), do: :eip161_test
  defp get_test("Byzantium"), do: :byzantium_test
  defp get_test("Constantinople"), do: :constantinople_test

  defp get_test("ByzantiumToConstantinopleAt5"),
    do: :byzantium_to_constantinople_transition_test

  defp get_test("EIP158ToByzantiumAt5"),
    do: :eip158_to_byzantium_transition_test

  defp get_test("HomesteadToDaoAt5"), do: :dao_hardfork_test
```
Is this worse too?

**Pull request merged**